scipy/stats/_multicomp.py

```diff
@@ -0,0 +1,311 @@
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import TYPE_CHECKING
+
+import numpy as np
+from numpy.testing import suppress_warnings
+
+from scipy import stats
+from scipy.optimize import minimize_scalar
+from scipy.stats._common import ConfidenceInterval
+from scipy.stats._qmc import check_random_state
+
+if TYPE_CHECKING:
+    import numpy.typing as npt
+    from scipy._lib._util import DecimalNumber, SeedType
+    from typing import Literal
+
+
+__all__ = [
+    'dunnett'
+]
+
+
+@dataclass
+class DunnettResult:
+    statistic: np.ndarray
+    pvalue: np.ndarray
+    _rho: np.ndarray
+    _df: int
+    _std: float
+    _observations_mean: np.ndarray
+    _control_mean: np.ndarray
+
+    def _allowance(self, confidence_level: DecimalNumber = 0.95) -> float:
+        """Allowance.
+
+        It is the quantity to add/substract from the observed difference
+        between the means of observed groups and the mean of the control
+        group. The result gives confidence limits.
+
+        Parameters
+        ----------
+        confidence_level : float, optional
+            Confidence level for the computed confidence interval
+            of the estimated proportion. Default is .95.
+
+        Returns
+        -------
+        allowance : float
+            Allowance around the mean.
+        """
+        dist = stats.multivariate_t(shape=self._rho, df=self._df)
+        alpha = 1 - confidence_level
+
+        def pvalue_from_stat(statistic):
+            # two-sided to have +- bounds
+            sf = 1 - dist.cdf(statistic, lower_limit=-statistic)
+            return abs(sf - alpha)
+
+        # scipy.stats.sampling methods are not working for this distribution
```

Comment:
```suggestion
        # scipy.stats.sampling does not support multivariate distributions
```
Although you could wrap it and this would effectively be a univariate distribution because the CDF is effectively a parameter of one variable, `statistic`.

Reply:
What I mean is that I did try that and it was either too slow that I killed the process (`NumericalInverseHermite`) or just wrong output (`NumericalInversePolynomial`)

```python
dist = multivariate_t(shape=rho, df=df)
dist_new = NumericalInversePolynomial(dist)
dist_new.ppf(0.95)
# 1.3236341183131461
dist.cdf(1.323634455420128)
# 0.7666263121306259
```

Reply:
The code above passes a frozen `multivariate_t` into `NumericalInversePolynomial`. We wouldn't expect that to work.

- `NumericalInversePolynomial` doesn't work with multivariate distributions 
- Even if we were to wrap the distribution to mimic a univariate distribution - we can't easily give `NumericalInversePolynomial` the univariate PDF it needs.

We could give `NumericalInverseHermite` an object with univariate CDF, maybe something like:

```python3
        class Dist:
            def __init__(self, rho, df):
                self.dist = stats.multivariate_t(shape=rho, df=df)
                self.dim = np.shape(rho)[0]

            def cdf(self, x):
                # needs `lower_limit` for 'two-sided'
                return self.dist.cdf(np.full(self.dim, fill_value=x),
                                     random_state=np.random.default_rng(0))

        dist = Dist(self._rho, self._df)
        #  The CDF is calculated with QMC integration, so it is difficult
        # to get the u-error below the default threshold. 
        dist_new = sampling.NumericalInverseHermite(dist, u_resolution=1e-4)
```
But unless there are many `alpha` for which we need the corresponding statistic, it would not be efficient to evaluate the CDF the hundreds or thousands of times needed to form the cubic Hermite spline. We might as well just evaluate the CDF the handful of times required by `minimize_scalar`.

My point is just that the comment should be adjusted. "scipy.stats.sampling methods are not working for this distribution" suggests that you would expect them to work, but they are not. But we shouldn't expect them to work.

It's probably best to just remove the comment, since using a scalar rootfinder or minimizer is the natural choice anyway.
```suggestion
```

Reply:
I see ok thanks. Let's remove the comment then.